examples/application/evaluate_gaming_traffic.cc

```diff
@@ -56,49 +49,51 @@ GeneratedPacketsStats (Ptr<OutputStreamWrapper> stream, Ptr<const Packet> packet
+  destination->GetTotalReceivedPackets ();
+  Time now = Simulator::Now ();  /* Return the simulator's virtual time. */
+  double cur = (destination->GetTotalReceivedBytes () - lastTotalRx) * 8.0 / computeThroughputPeriodicity.GetSeconds () / 1e3;  /* Convert Application RX Packets to kb */
   *stream->GetStream () << now.GetSeconds ()  << "," << cur << std::endl;
-  lastTotalRx = packetSink->GetTotalRx ();
-  Simulator::Schedule (computeThroughputPeriodicity, &CalculateThroughput, stream);
+  lastTotalRx = destination->GetTotalReceivedBytes ();
+  Simulator::Schedule (computeThroughputPeriodicity, &CalculateThroughput, stream, destination, lastTotalRx);
 }
 
 int
 main (int argc, char *argv[])
 {
-  bool summary = true;                                /* Print application layer traffic summary */
-  double bitrate = 0;                                 /* Application data rate (in MBps) */
-  double simulationTime = 10.0;                       /* Simulation time in seconds */
-  std::string game = "ns3::CrazyTaxiStreamingServer"; /* TypeId of the game */
+  Ptr<GamingStreamingServer> gamingServer;     /* Pointer to the gaming server application */
+  Ptr<GamingStreamingServer> gamingClient;     /* Pointer to the gaming server application */
+  uint64_t serverLastTotalRx = 0;              /* The value of the last total received bytes */
+  uint64_t clientLastTotalRx = 0;              /* The value of the last total received bytes */
+  Time serverLastPacketTime = Seconds (-1.0);  /* Time of last packet generated by server */
+  Time clientLastPacketTime = Seconds (-1.0);  /* Time of last packet generated by client */
+
+  bool summary = true;                      /* Print application layer traffic summary */
+  double serverBitrate = 0;                 /* Gaming server data rate (in MBps) */
+  double clientBitrate = 0;                 /* Gaming client data rate (in MBps) */
+  double simulationTime = 10.0;             /* Simulation time in seconds */
+  std::string gamingServerId = "ns3::CrazyTaxiStreamingServer"; /* TypeId of the gaming server */
+  std::string gamingClientId = "ns3::CrazyTaxiStreamingClient"; /* TypeId of the gaming server */
 
   CommandLine cmd;
   cmd.AddValue ("summary", "Print summary of application layer traffic", summary);
-  cmd.AddValue ("bitrate", "Application data rate (in MBps)", bitrate );
+  cmd.AddValue ("server-bitrate", "Gaming server data rate (in MBps)", serverBitrate );
+  cmd.AddValue ("client-bitrate", "Gaming client data rate (in MBps)", clientBitrate );
   cmd.AddValue ("time", "Simulation time (in Seconds)", simulationTime );
   cmd.AddValue ("throughput", "Period in which throughput calculated", computeThroughputPeriodicity );
-  cmd.AddValue ("game", "The game TypeId", game );
+  cmd.AddValue ("game-server", "The gaming server TypeId", gamingServerId );
+  cmd.AddValue ("game-client", "The gaming client TypeId", gamingClientId );
```

Comment:
i would bind the of them together with a single input, e.g., gamingServerType={CrazyTaxi, FourElements}